mesonbuild/compilers/d.py

```diff
@@ -0,0 +1,318 @@
+                # a linker search path.
+                dcargs.append('-L' + arg)
+                continue
+            dcargs.append(arg)
+
+        return dcargs
+
+
+class GnuDCompiler(DCompiler):
+    def __init__(self, exelist, version, is_cross):
+        DCompiler.__init__(self, exelist, version, is_cross)
+        self.id = 'gcc'
+        default_warn_args = ['-Wall', '-Wdeprecated']
+        self.warn_args = {'1': default_warn_args,
+                          '2': default_warn_args + ['-Wextra'],
+                          '3': default_warn_args + ['-Wextra', '-Wpedantic']}
+        self.base_options = ['b_colorout', 'b_sanitize', 'b_staticpic']
+
+    def get_colorout_args(self, colortype):
+        if mesonlib.version_compare(self.version, '>=4.9.0'):
+            return gnu_color_args[colortype][:]
+        return []
+
+    def get_dependency_gen_args(self, outtarget, outfile):
+        return ['-fmake-deps=' + outfile]
+
+    def get_output_args(self, target):
+        return ['-o', target]
+
+    def get_linker_output_args(self, target):
+        return ['-o', target]
+
+    def get_include_args(self, path, is_system):
+        return ['-I' + path]
+
+    def get_warn_args(self, level):
+        return self.warn_args[level]
+
+    def get_werror_args(self):
+        return ['-Werror']
+
+    def get_linker_search_args(self, dirname):
+        return ['-L' + dirname]
+
+    def get_buildtype_args(self, buildtype):
+        return d_gdc_buildtype_args[buildtype]
+
+    def build_rpath_args(self, build_dir, rpath_paths, install_rpath):
+        return build_unix_rpath_args(build_dir, rpath_paths, install_rpath)
+
+    def get_unittest_args(self):
+        return ['-funittest']
+
+
+class LLVMDCompiler(DCompiler):
+    def __init__(self, exelist, version, is_cross):
+        DCompiler.__init__(self, exelist, version, is_cross)
+        self.id = 'llvm'
+        self.base_options = ['b_coverage', 'b_colorout']
+
+    def get_colorout_args(self, colortype):
+        if colortype == 'always':
+            return ['-enable-color']
+        return []
+
+    def get_dependency_gen_args(self, outtarget, outfile):
+        # LDC using the -deps flag returns a non-Makefile dependency-info file, which
+        # the backends can not use. So we disable this feature for now.
+        return []
+
+    def get_output_args(self, target):
+        return ['-of', target]
+
+    def get_linker_output_args(self, target):
+        return ['-of', target]
+
+    def get_include_args(self, path, is_system):
+        return ['-I' + path]
+
+    def get_warn_args(self, level):
+        if level == '2' or level == '3':
+            return ['-wi', '-dw']
+        else:
+            return ['-wi']
+
+    def get_werror_args(self):
+        return ['-w']
+
+    def get_coverage_args(self):
+        return ['-cov']
+
+    def get_buildtype_args(self, buildtype):
+        return d_ldc_buildtype_args[buildtype]
```

Comment:
__[flake8]__
* [F405] 'd_ldc_buildtype_args' may be undefined, or defined from star imports: .compilers

<!-- Comment by [SideCI](https://sideci.com) -->